src/Microsoft.AspNetCore.Http/Internal/DefaultHttpRequest.cs

```diff
@@ -41,7 +41,7 @@ public virtual void Uninitialize()
             _features.Fetch(ref _features.Cache.Query, f => new QueryFeature(f));
 
         private IFormFeature FormFeature =>
-            _features.Fetch(ref _features.Cache.Form, this, f => new FormFeature(f));
+            _features.Fetch(ref _features.Cache.Form, this, f => new FormFeature(f, (f._context as DefaultHttpContext)?.FormOptions));
```

Comment:
Just hopping in, but isn't tying this class to `DefaultHttpContext` this way a little ugly? Why not flow `FormOptions` through the constructor when initializing it like you [flow it](https://github.com/aspnet/HttpAbstractions/pull/675/files#diff-b5cd72dfa894e4474c6091bc1a696735R46) from `DefaultHttpContextFactory` to `DefaultHttpContext`?

Same goes for `DefaultHttpResponse` with the `BuilderPool` property.